node/runtime/src/lib.rs

```diff
@@ -784,9 +787,7 @@ impl Runtime {
                 }
             };
         } else {
-            // wrong receipt
-            result.status = TransactionStatus::Failed;
-            result.logs.push("receipt sent to the wrong shard".to_string());
+            unreachable!("receipt sent to the wrong shard");
```

Comment:
I don't think should be unreachable. Could be a malicious behavior.

Reply:
How would you react to this malicious behavior? If it's possible to spoof receipts, then there is no security of the network.
It's either should be panic or unreachable.

Reply:
I see